filedialog.cpp

```diff
@@ -42,6 +36,23 @@ Dialog::Dialog(QWidget *parent, const QString &filename, int mode)
     QObject::connect(m_filedlg, SIGNAL(fileSelected(QString)), this, SLOT(selectFile(QString)));
 }
 
+void Dialog::currentFile(QString file)
+{
+    m_fileinfo->setFile(file);
+    if (m_fileinfo->isFile())
+    {
+        QString info;
+        info = "FilePath: \n" + m_fileinfo->absoluteFilePath() + "\n" + "FileName: \n" + m_fileinfo->fileName();
+        m_mediainfo->setFontPointSize(12);
+        m_mediainfo->setText(info);
+    }
+}
+
+void Dialog::selectFile(QString file)
+{
```

Comment:
**Suggestion:** The `Dialog` class should be updated to handle the current file selection more effectively, such as displaying file information in a `QTextEdit` widget. [enhancement, importance: 6]
```suggestion
void Dialog::selectFile(QString file)
{
    m_file = file;
    currentFile(file);
}
void Dialog::currentFile(QString file)
{
    m_fileinfo->setFile(file);
    if (m_fileinfo->isFile())
    {
        QString info = "FilePath: \n" + m_fileinfo->absoluteFilePath() + "\n" + "FileName: \n" + m_fileinfo->fileName();
        m_mediainfo->setFontPointSize(12);
        m_mediainfo->setText(info);
    }
}
```